docs/developer.md

```diff
@@ -1,112 +0,0 @@
-## Developing the Provider
```

Comment:
Why remove this ?

Reply:
I'm just running this 

```
tfplugindocs:
	go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
	tfplugindocs generate
```

the latest binary removes the file